### PR TITLE
Rplidar with sdk

### PR DESCRIPTION
**Pull request by aerydna**
Implementation of new rplidar2 device via productor sdk.

### CODE REVIEW

File: src/devices/rpLidar2/sdk/src/arch/macOS/net_serial.cpp [hunk 1]
```diff
@@ -0,0 +1,347 @@
+/*
```
Comment (randaz81) [inline src/devices/rpLidar2/sdk/src/arch/macOS/net_serial.cpp]: rejected, cannot be modified.

File: src/devices/rpLidar2/sdk/src/arch/linux/net_serial.cpp [hunk 2]
```diff
@@ -0,0 +1,343 @@
+/*
```
Comment (randaz81) [inline src/devices/rpLidar2/sdk/src/arch/linux/net_serial.cpp]: rejected, cannot be modified.

File: src/devices/rpLidar2/sdk/src/rplidar_driver.cpp [hunk 3]
```diff
@@ -0,0 +1,1026 @@
...
+            return RESULT_OPERATION_FAIL;
+        }
+    }
+    return RESULT_OK;
+}
+
+
+u_result RPlidarDriverSerialImpl::startScan(bool force, bool autoExpressMode)
+{
+    bool isExpressModeSupported;
```
Comment (randaz81) [inline src/devices/rpLidar2/sdk/src/rplidar_driver.cpp]: rejected, cannot be modified.

File: src/devices/rpLidar2/sdk/src/rplidar_driver.cpp [hunk 4]
```diff
@@ -0,0 +1,1026 @@
...
+void RPlidarDriver::DisposeDriver(RPlidarDriver * drv)
+{
+    delete drv;
+}
+
+
+
+// Serial Driver Impl
+
+RPlidarDriverSerialImpl::RPlidarDriverSerialImpl()
```
Comment (randaz81) [inline src/devices/rpLidar2/sdk/src/rplidar_driver.cpp]: rejected, cannot be modified.

File: src/devices/rpLidar2/sdk/src/arch/linux/thread.hpp [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    case PRIORITY_NORMAL:
+    case PRIORITY_LOW:
+    case PRIORITY_IDLE:
+        //pthread_priority = 0;
+        current_policy = SCHED_OTHER;
+        break;
+    }
+
+    current_param.__sched_priority = current_policy;
+    if ( (ans = pthread_setschedparam( (pthread_t) this->_handle, current_policy, &current_param)) )
```
Comment (randaz81) [inline src/devices/rpLidar2/sdk/src/arch/linux/thread.hpp]: rejected, cannot be modified.

**Comment by traversaro** [comment]
FYI : The Codacy check can be disabled in selected directories in https://www.codacy.com/app/robotology/yarp/settings/ignoreFiles .

**Comment by drdanz** [comment]
@traversaro True, but unfortunately I don't think you can disable it for files that are not yet in the repository.
On the other hand, the external folder is already disabled, @aerydna, @randaz81 I suggest to move the rplidar sdk there...

**Comment by drdanz** [comment]
@aerydna please move the SDK in the external folder

**Comment by aerydna** [comment]
sdk moved!

**Comment by aerydna** [comment]
codacy error fixed. should be ready to be merged

**Pull request merged**